### PR TITLE
fix: Signing out clearing all user preferences

### PR DESCRIPTION
**Pull request by helpisdev**
Fixes a bug (unintended behavior) where performing a 'sign out with google' action, clears all user's shared preferences, even values not used by the plugin. Fixed it by just setting to null the values the plugin actually uses, since it already had a `clear` method that almost did exactly that, except for the user ID.

### CODE REVIEW

**Comment by ihorkozar** [comment]
@long1eu Hi, any updates on this?

File: example/lib/main.dart [hunk 1]
```diff
@@ -1,6 +1,8 @@
-// Copyright 2019 The Flutter Authors. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
+// Copyright (C) Hellenic Progressive Internet Services, Inc.
+// All Rights Reserved. 2022.
+// Unauthorized copying of this file, via any medium is strictly prohibited.
+// Proprietary and confidential.
+// Written by Elias Kapareliotis <helpis@tutamail.com>.
```
Comment (long1eu) [inline example/lib/main.dart:5]: Why? @helpisdev

File: example/lib/main.dart [hunk 2]
```diff
@@ -11,19 +13,17 @@ import 'package:extension_google_sign_in_as_googleapis_auth/extension_google_sig
...
 import 'package:google_sign_in/google_sign_in.dart';
+import 'package:google_sign_in_dart_example/platform_js.dart'
+    if (dart.library.io) 'platform_io.dart';
 import 'package:google_sign_in_dartio/google_sign_in_dartio.dart';
 import 'package:googleapis/gmail/v1.dart';
 import 'package:googleapis/people/v1.dart';
 import 'package:googleapis_auth/googleapis_auth.dart';
 import 'package:html_unescape/html_unescape.dart';
 
-import 'platform_js.dart' if (dart.library.io) 'platform_io.dart';
```
Comment (long1eu) [inline example/lib/main.dart:20]: please explain

File: lib/google_sign_in_dartio.dart [hunk 3]
```diff
@@ -29,37 +34,45 @@ typedef UrlPresenter = void Function(Uri uri);
...
+  /// login succeeds or fails
   static Future<void> register({
-    required String clientId,
-    String? exchangeEndpoint,
+    required final String clientId,
+    final String? exchangeEndpoint,
     DataStorage? storage,
     UrlPresenter? presenter,
-    int? port,
+    final int? port,
```
Comment (long1eu) [inline lib/google_sign_in_dartio.dart:64]: I think that we need to have an Options class that we should pass and not keep adding new values here.

**Comment by long1eu** [comment]
> @long1eu Hi, any updates on this?

waiting for some clarifications

**Comment by helpisdev** [comment]
Yes I'm sorry Android Studio config, I was just checking that. Seems that I had forgotten I had opened the pull request and while working on another private project things got pushed together, my mistake. I checked the git history and I don't see to have changed anything important after the merging other open prs. All the garbage commits are after that point. Since it's linear history up to there, theoretically a git hard reset should solve the issue. I'm currently working on it. Sorry for the inconvenience this happened due to my negligence.

**Comment by helpisdev** [comment]
Again, I'm really sorry for the trouble, I feel kinda ashamed I didn't notice that earlier. Git reset hard and git push -f restored it to clear point and removed all the unnecessary garbage commits.

**Comment by long1eu** [comment]
Absolutely no problem. Thank you for your interest.

**Comment by long1eu** [comment]
Please recreate and let me know.

**Pull request closed without merging**